actionners/kubernetes/drain/drain.go

```diff
@@ -0,0 +1,257 @@
+package drain
+
+import (
+	"context"
+	"fmt"
+	"github.com/falco-talon/falco-talon/internal/events"
+	kubernetes "github.com/falco-talon/falco-talon/internal/kubernetes/client"
+	"github.com/falco-talon/falco-talon/internal/rules"
+	"github.com/falco-talon/falco-talon/utils"
+	"github.com/go-playground/validator/v10"
+	policyv1 "k8s.io/api/policy/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	validatorName = "is_absolut_or_percent"
+	// EvictionKind represents the kind of evictions object
+	EvictionKind = "Eviction"
+	// EvictionSubresource represents the kind of evictions object as pod's subresource
+	EvictionSubresource = "pods/eviction"
+)
+
+type Config struct {
+	MinHealthyReplicas string `mapstructure:"min_healthy_replicas" validate:"omitempty,is_absolut_or_percent"`
+	IgnoreDaemonsets   bool   `mapstructure:"ignore_daemonsets" validate:"omitempty"`
+	IgnoreStatefulSets bool   `mapstructure:"ignore_statefulsets" validate:"omitempty"`
+	GracePeriodSeconds int    `mapstructure:"grace_period_seconds" validate:"omitempty"`
+}
+
+func Action(action *rules.Action, event *events.Event) (utils.LogLine, error) {
+	podName := event.GetPodName()
+	namespace := event.GetNamespaceName()
+
+	objects := map[string]string{
+		"pod":       podName,
+		"namespace": namespace,
+	}
+
+	parameters := action.GetParameters()
+	gracePeriodSeconds := new(int64)
+	if parameters["grace_period_seconds"] != nil {
+		*gracePeriodSeconds = int64(parameters["grace_period_seconds"].(int))
+	}
+
+	client := kubernetes.GetClient()
+	pod, err := client.GetPod(podName, namespace)
+	if err != nil {
+		objects["pod"] = podName
+		objects["namespace"] = namespace
+		return utils.LogLine{
+				Objects: objects,
+				Error:   err.Error(),
+				Status:  "failure",
+			},
+			err
+	}
+
+	node, err := client.GetNodeFromPod(pod)
+	if err != nil {
+		return utils.LogLine{
+				Objects: objects,
+				Error:   err.Error(),
+				Status:  "failure",
+			},
+			err
+	}
+
+	objects["node"] = node.Name
+
+	pods, err := client.Clientset.CoreV1().Pods("").List(context.Background(), metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("spec.nodeName=%s", node.GetName()),
+	})
+
+	for _, p := range pods.Items {
+
+		if parameters["ignore_daemonsets"] != nil || parameters["ignore_statefulsets"] != nil || parameters["min_healthy_replicas"] != nil {
+			pod, err := client.GetPod(p.Name, p.Namespace)
+			if err != nil {
+				return utils.LogLine{
+						Objects: objects,
+						Error:   err.Error(),
+						Status:  "failure",
+					},
+					err
+			}
+
+			if len(pod.OwnerReferences) != 0 {
+				switch pod.OwnerReferences[0].Kind {
+				case "DaemonSet":
+					if parameters["ignore_daemonsets"].(bool) {
+						return utils.LogLine{
```

Comment:
No return here, we just omit to call the eviction endpoint for this pod, same for other cases